build-tools/xaprepare/xaprepare/Application/Program.ArchLinux.cs

```diff
@@ -1,29 +1,139 @@
 using System;
+using System.Collections.Generic;
+using System.IO;
 using System.Threading.Tasks;
 
 namespace Xamarin.Android.Prepare
 {
-	class ArchLinuxProgram : LinuxProgram
-	{
-		public ArchLinuxProgram (string packageName, string executableName = null)
-			: base (packageName, executableName)
-		{}
+    class ArchLinuxProgram : LinuxProgram
+    {
+        private static string aurHelper = null;
+		// TODO: check if binary name is right.
+        private static string[] aurHelpersToCheck = new string[] { "yay","aurman","pacaur","pakku","aura","pikaur","trizen","pbget","auracle","repofish","pb", "bb", "rua" };
+        /// <summary>
+        /// Available AUR helper which has to support "-Q", "-S".
+        /// </summary>
+        public static string AvailableAurHelper
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(aurHelper))
+                {
+                    return null;
+                }
+                return aurHelper = CheckForAvailability(aurHelpersToCheck);
+            }
+        }
 
-		protected override bool CheckWhetherInstalled ()
+        private static string CheckForAvailability(string[] aurHelpersToCheck)
+        {
+			foreach (var item in aurHelpersToCheck)
+			{
+				if(ExistsOnPath(item))
+					return item;
+			}
+			return null;
+        }
+
+		#region Check for executable
+		// code from https://stackoverflow.com/a/3856090
+		private static bool ExistsOnPath(string fileName)
 		{
-			throw new NotImplementedException ();
+			return GetFullPath(fileName) != null;
 		}
 
-#pragma warning disable CS1998
-		public override async Task<bool> Install ()
+		private static string GetFullPath(string fileName)
 		{
-			throw new NotImplementedException ();
+			if (File.Exists(fileName))
+				return Path.GetFullPath(fileName);
+
+			var values = Environment.GetEnvironmentVariable("PATH");
+			foreach (var path in values.Split(Path.PathSeparator))
+			{
+				var fullPath = Path.Combine(path, fileName);
+				if (File.Exists(fullPath))
+					return fullPath;
+			}
+			return null;
```

Comment:
You don't need this code, you should instead use the [OS.Which](https://github.com/xamarin/xamarin-android/blob/master/build-tools/xaprepare/xaprepare/OperatingSystems/OS.cs#L453-L464) method via `Context.OS` (or `Context.Instance.OS`, as applicable).